integration-tests/jiva/invalidconfig/invalidconfig_test.go

```diff
@@ -0,0 +1,194 @@
+	openebsProvisioner                = "openebs.io/provisioner-iscsi"
+	accessModes                       = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+	capacity                          = "5G"
+	scObj                             *storagev1.StorageClass
+	pvcObj                            *corev1.PersistentVolumeClaim
+	nsObj                             *corev1.Namespace
+	openebsCASConfigValue             = "- name: ReplicaCount:\n  Value: 3"
+	pvcLabel, replicaLabel, ctrlLabel string
+)
+
+var _ = Describe("TEST INVALID CAS CONFIGURATIONS IN SC", func() {
+	var (
+		namespaceName = "validation-ns1"
+		scName        = "jiva-invalid-config-sc"
+		pvcName       = "jiva-volume-claim"
+	)
+	BeforeEach(func() {
+		annotations := map[string]string{
+			string(apis.CASTypeKey):   string(apis.JivaVolume),
+			string(apis.CASConfigKey): openebsCASConfigValue,
+		}
+		var err error
+
+		nsObj, err = ns.NewBuilder().
+			WithName(namespaceName).
+			APIObject()
+		Expect(err).ShouldNot(HaveOccurred())
+
+		scObj, err = sc.NewBuilder().
+			WithName(scName).
+			WithAnnotations(annotations).
+			WithProvisioner(openebsProvisioner).Build()
+		Expect(err).ShouldNot(HaveOccurred())
+
+		pvcObj, err = pvc.NewBuilder().
+			WithName(pvcName).
+			WithNamespace(namespaceName).
+			WithStorageClass(scName).
+			WithAccessModes(accessModes).
+			WithCapacity(capacity).Build()
+		Expect(err).ShouldNot(HaveOccurred())
+
+		By(fmt.Sprintf("Creating test specific namespace {%s}", namespaceName))
+		_, err = ops.nsClient.Create(nsObj)
+		Expect(err).To(BeNil())
+
+		By(fmt.Sprintf("Creating PVC named {%s} in Namespace: {%s}", pvcName, namespaceName))
+		_, err = ops.pvcClient.WithNamespace(namespaceName).Create(pvcObj)
+		Expect(err).To(BeNil())
+
+		pvcLabel = string(apis.PersistentVolumeClaimKey) + "=" + pvcName
+		replicaLabel = defaultReplicaLabel + "," + pvcLabel
+		ctrlLabel = defaultCtrlLabel + "," + pvcLabel
+	})
+
+	When("We apply sc with invalid CASconfig yaml in k8s cluster", func() {
+		It("Jiva controller and replica pods should not create", func() {
+
+			By(fmt.Sprintf("Creating storageclass named {%s}", scName))
+			_, err := ops.scClient.Create(scObj)
+			Expect(err).To(BeNil())
+
+			By("jiva-ctrl pod should not come to running state")
+			podCount := ops.getPodCountRunningEventually(namespaceName, ctrlLabel, 1)
+			Expect(podCount).To(Equal(0))
+
+			By("jiva-replica pod should not come to running state")
+			podCount = ops.getPodCountRunningEventually(namespaceName, replicaLabel, 3)
+			Expect(podCount).To(Equal(0))
+		})
+	})
+
+	AfterEach(func() {
+		By(fmt.Sprintf("Delete PVC named {%s} in Namespace: {%s}", pvcName, namespaceName))
+		err := ops.pvcClient.Delete(pvcName, &metav1.DeleteOptions{})
+		Expect(err).To(BeNil())
+
+		By(fmt.Sprintf("Delete storageclass named {%s}", scName))
+		err = ops.scClient.Delete(scName, &metav1.DeleteOptions{})
+		Expect(err).To(BeNil())
+
+		By(fmt.Sprintf("Delete {%s} namespace", namespaceName))
```

Comment:
Please follow the same guidelines throughout.

Reply:
Updated the file PTAL